astropix.py

```diff
@@ -236,37 +302,71 @@ def init_injection(self, dac_settings:tuple[int, list[float]] = (2, [0.4, 0.0]),
 
     # These start and stop injecting voltage. Fairly simple.
     def start_injection(self):
+        """
+        Starts Injection.
+        Takes no arguments and no return
+        """
         self.injector.start()
-        print("BEGAN INJECTION")
+        logging.info("BEGAN INJECTION")
 
     def stop_injection(self):
+        """
+        Stops Injection.
+        Takes no arguments and no return
+        """
         self.injector.stop()
-        print("STOPPED INJECTION")
+        logging.info("STOPPED INJECTION")
 
 
 ########################### Input and Output #############################
     # This method checks the chip to see if a hit has been logged
 
     def hits_present(self):
+        """
+        Looks at interupt
+        Returns bool, True if present
+        """
         if (int.from_bytes(self.nexys.read_register(70),"big") == 0):
             return True
         else:
             return False
 
+    def get_log_header(self):
+        """
+        Returns header for use in a log file with all settings.
+        """
+        # This is not a nice line, but its the most efficent way to get all the values in the same place.
+        return f"Voltageboard settings: {self.vboard.dacvalues}\n" + f"Digital: {self.digitalconfig}\n" +f"Biasblock: {self.bias_setup}\n" + f"DAC: {self.dac_setup}\n" + f"Receiver: {self.recconfig}\n"
+
 
 ############################ Decoder Stuffs ##############################
     # This function generates a list of the hits in the stream. Retuerns a bytearray
 
-    def get_readout(self, return_hex: bool = False):
+    def get_readout(self):
+        """
+        Reads hit buffer.
+        No arguments. 
+        Returns bytearray
+        """
         self.nexys.write_spi_bytes(20)
         readout = self.nexys.read_spi_fifo()
-        if return_hex:
-            return binascii.hexlify(readout)
-        else:
-            return readout
+        return readout
 
 
```

Comment:
function to print hexidecimal info?

Reply:
I moved that into the run script. If you think it would be helpful I can add a `printer:bool = False` argument which would print the hex string to stdout

Reply:
I like the way you've done it :)